### PR TITLE
add sequelize-cursor-pagination type declaration

### PR DESCRIPTION
**Pull request by pilagod**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/sequelize-cursor-pagination/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        raw?: boolean;
+    }
+
+    interface WithPaginationOptions<MethodName extends string> {
+        methodName?: MethodName; // [default: 'paginate']
+        primaryKeyField?: string; // [default: 'id']
+    }
+
+    type WithPaginationModel<MethodName extends string, Model, ModelInst, ModelAttrs> = Model & (
+        {
```
Comment (DanielRosenwasser) [inline types/sequelize-cursor-pagination/index.d.ts]: Can't you avoid the intersection and just write this as a type with two overloads?

```
{
  [key in MethodName]: {
    (options?: PaginateRawOptions<ModelAttrs>): { results: ModelAttrs[], cursors: Cursors }
    (options?: PaginateInstOptions<ModelAttrs>): { results: ModelInst[], cursors: Cursors }
  }
}
```

Comment (pilagod) [reply types/sequelize-cursor-pagination/index.d.ts]: I didn't realize this refactoring ! This makes the type more clear, many thanks 😃

File: types/sequelize-cursor-pagination/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+declare namespace SequelizeCursorPagination {
+    interface Cursors {
+        before: string | null;
+        after: string | null;
+        hasNext: boolean;
+        hasPrevious: boolean;
+    }
+
+    interface BasicPaginateOptions<T> {
```
Comment (pilagod) [inline types/sequelize-cursor-pagination/index.d.ts]: I come up a refactoring here:
```
where?: Sequelize.FindOptions<T>['where'];
attributes?: Sequelize.FindOptions<T>['attributes'];
include?: Sequelize.FindOptions<T>['include'];
```
Is it a good practice ? Or should I do by extending & Pick:
```
interface BasicPaginateOptions<T> extends Pick<Sequelize.FindOptions<T>, 'where' | 'attributes' | 'include'> {
    ...
} 
```

Comment (DanielRosenwasser) [reply types/sequelize-cursor-pagination/index.d.ts]: I think either should work fine. If you'd like, you can send a follow-up PR. Thanks!

Comment (pilagod) [reply types/sequelize-cursor-pagination/index.d.ts]: No problem, many thanks for your review 🙂

**Pull request merged**